src/sdkFactory/index.ts

```diff
@@ -65,7 +69,10 @@ export function sdkFactory(params: ISdkFactoryParams): SplitIO.ICsSDK | SplitIO.
   const integrationsManager = integrationsManagerFactory && integrationsManagerFactory({ settings, storage });
 
   const observer = impressionsObserverFactory();
-  const strategy = (storageFactoryParams.optimize) ? strategyOptimizedFactory(observer, storage.impressionCounts!) : strategyDebugFactory(observer);
+  const filterAdapter = filterAdapterFactory && filterAdapterFactory();
+  const uniqueKeysTracker = uniqueKeysTrackerFactory(log, filterAdapter);
+  const strategy = (storageFactoryParams.optimize) ? strategyOptimizedFactory(observer, storage.impressionCounts!) :
```

Comment:
As we discussed today, for future polishing, lets refactor `IStorageFactoryParams` interface, by passing the settings object, to remove `optimize`, and other props that are already present in the settings.

Reply:
Added to checklist 📝